src/main/java/org/sobotics/guttenberg/clients/Guttenberg.java

```diff
@@ -182,4 +198,49 @@ private void checkLastExecution() {
 		}
 		
 	}
+	
+	/**
+	 * Checks for updates
+	 * */
+	private void update() {
+		System.out.println("Load updater...");
+		Updater updater = new Updater();
+		System.out.println("Check for updates...");
+		
+		int update = updater.updateIfAvailable(); 
+		
+		if (update == -1) {
+			for (Room room : this.chatRooms) {
+				if (room.getRoomId() == 111347) {
+					room.send("Automatic update failed!");
+				}
+			}
+		} else if (update == 1) {
+			for (Room room : this.chatRooms) {
+				if (room.getRoomId() == 111347) {
+					room.send("Rebooting for update to version "+updater.getNewVersion().get());
+				}
+				room.leave();
+			}
+			System.exit(0);
+		}
+		
+	}
+	
+	
+	
+	private void setLogfile() {
+		Date now = Date.from(Instant.now());
+		String dateString = new SimpleDateFormat("yyyy-MM-dd-HH").format(now);
+		
+		try {
+			PrintStream stream = new PrintStream(new FileOutputStream("./logs/guttenberg_"+dateString+".txt"));
+			System.setOut(stream);
```

Comment:
Instead of overriding `System.out`, prefer to use a real logger. Add a dependency of Log4J for example, and configure it to log to a file. Much better.

1. Add a dependency on `slf4j-log4j12` in the POM (because chatexchange uses the Slf4j facade, but you could use any other)

```xml
<dependency>
    <groupId>org.slf4j</groupId>
    <artifactId>slf4j-log4j12</artifactId>
    <version>1.7.21</version>
</dependency>
```

2. Use a logger in your class with

```java
private static final Logger LOGGER = LoggerFactory.getLogger(MyClass.class);
// ...
LOGGER.info("Plopz");
```